examples/How_to_use_moderation.ipynb

```diff
@@ -0,0 +1,455 @@
+    "\n",
+    "#### Workflow:\n",
+    "We will create a workflow that incorporates the Moderation API to check the LLM response for any potentially unsafe content before it is sent to the Language Model (LLM). This ensures that only appropriate content is displayed to the user.\n",
+    "\n",
+    "1. Receive an input from the user\n",
+    "2. Send prompt to LLM and generate a response\n",
+    "3. Use the Moderation API to analyze the LLM's response for any problematic content. \n",
+    "3. **Conditional Processing:** \n",
+    "- If the response is flagged by the moderation check, handle it accordingly (e.g., reject the response, show a placeholder message, etc.).\n",
+    "- If the response is not flagged, display it to the user.\n",
+    "\n",
+    "We will demonstrate this workflow with two example prompts:\n",
+    "\n",
+    "- **Bad Prompt:** An example input that should trigger the moderation categories and be blocked from reaching the user.\n",
+    "- **Good Prompt:** An example input that should pass the moderation check and be sent back to the user as a valid response."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "async def execute_all_moderations(user_request):\n",
+    "    input_moderation_task = asyncio.create_task(check_expression(user_request))\n",
+    "    chat_task = asyncio.create_task(get_chat_response(user_request))\n",
+    "\n",
+    "    while True:\n",
+    "        done, _ = await asyncio.wait(\n",
+    "            [input_moderation_task, chat_task], return_when=asyncio.FIRST_COMPLETED\n",
+    "        )\n",
+    "        if input_moderation_task in done:\n",
+    "            input_moderation_response = input_moderation_task.result()\n",
+    "            if input_moderation_response == True:\n",
+    "                chat_task.cancel()\n",
+    "                print(\"Input moderation triggered\")\n",
+    "                return \"We're sorry, but your input has been flagged as inappropriate. Please rephrase your input and try again.\"\n",
+    "            elif chat_task in done:\n",
+    "                chat_response = chat_task.result()\n",
+    "                output_moderation_response = await check_expression(chat_response)\n",
+    "\n",
+    "                if output_moderation_response == True:\n",
+    "                    print(f\"Moderation flagged for LLM response.\")\n",
+    "                    return \"Sorry, we're not permitted to give this answer. I can help you with any general queries you might have.\"\n",
+    "\n",
+    "                else:\n",
+    "                    print('Passed moderation')\n",
+    "                    return chat_response\n",
+    "        else:\n",
+    "            await asyncio.sleep(0.1)"
+   ]
```

Comment:
Same general critique as the first code section. One specific examples is, instead of doing
```python
if input_moderation in done:
  [happy path]
else:
  [wait]
```
I would do
```python
if input_moderation not in done:
  [wait]
  continue # short circuits
[happy path] # now this is no longer nested
```

Reply:
done